app/models/work_package/pdf_export/overview_table.rb

```diff
@@ -0,0 +1,178 @@
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module WorkPackage::PDFExport::OverviewTable
+  def write_work_packages_overview!(work_packages, id_wp_meta_map)
+    if query.grouped?
+      write_grouped!(work_packages, id_wp_meta_map)
+    else
+      with_margin(overview_table_margins_style) do
+        write_table!(work_packages, id_wp_meta_map, get_total_sums)
+      end
+    end
+  end
+
+  private
+
+  def write_grouped!(work_packages, id_wp_meta_map)
+    groups = {}
+    work_packages.each do |work_package|
+      group = query.group_by_column.value(work_package)
+      groups[group] = groups[group] || []
+      groups[group].push work_package
+    end
+    groups.each do |group, grouped_work_packages|
+      write_group!(group, grouped_work_packages, id_wp_meta_map, get_group_sums(group))
+    end
+  end
+
+  def table_columns_objects
+    @table_columns_objects ||= limit_table_columns_objects
+  end
+
+  def limit_table_columns_objects
+    list = column_objects
+    list = list.reject { |c| c == query.group_by_column } if query.grouped?
+    list = list.take(4) if with_descriptions?
+    list
+  end
+
+  def table_column_widths
+    widths = table_columns_objects.map do |col|
+      col.name == :subject || text_column?(col) ? 4.0 : 1.0
+    end
+    widths.unshift 1.0 if with_descriptions?
+    ratio = pdf.bounds.width / widths.sum
+    widths.map { |w| w * ratio }
+  end
+
+  def write_group!(group, work_packages, id_wp_meta_map, sums)
+    write_optional_page_break(page_break_space_left_threshold)
+    with_margin(overview_table_margins_style) do
+      label = make_group_label(group)
+      with_margin(overview_group_header_margins_style) do
+        pdf.formatted_text([overview_group_header_style.merge({ text: label })])
+      end
+      write_table!(work_packages, id_wp_meta_map, sums)
+    end
+  end
+
+  def write_table!(work_packages, id_wp_meta_map, sums)
+    rows = build_table_rows(work_packages, id_wp_meta_map, sums)
+    pdf_table_auto_widths(rows, table_column_widths, table_options) do |table|
+      format_sum_cells table.cells.columns(0..-1).rows(-1) if query.display_sums?
+    end
+  end
+
+  def table_options
+    { header: true, cell_style: overview_table_cell_style.merge({ inline_format: true }) }
+  end
+
+  def format_sum_cells(sums_cells)
+    sums_style = overview_table_sums_cell_style
```

Comment:
:thinking: I was a bit confused here, where this method was coming from. Again, we are expecting that the `Style` module is included as well and that this defined the method. Maybe in the Future we can also there make those method module scoped methods and instead use `Style.overview_table_sums_cell_style`, so we know where to look for the definition of the method.

Reply:
This is what I am working on atm for a second PR, all these calls will be like `style.overview_table_sums_cell`, so it will much clearer where they come from.